### PR TITLE
tgui-next: Low level stuff

### PR DESCRIPTION
**Pull request by stylemistake**
## About The Pull Request

- Improve build times (22.7s -> 13.8s)
- Physically separate development and production builds so that they never overwrite each other.
  - Move production builds one level up in public folder
  - Gitignore development builds
- Warn about lowercase onclick handlers.
    - This was a common source of bugs on some russian tgstation fork, educational warnings are always good.
- Use camelcase `tabIndex` on buttons.
    - Performance improvement mostly
- Improve byond cache path searching,
    - Faster and more optimal cache reloading.
- Add WSL-specific paths to cache reloader.
    - Official out of the box support for WSL, no more symlinkage.
- Use `url-loader` for inlining images as base64.
    - No more manual inlining of SVGs in CSS.
- Reload once by calling `bin/tgui --dev --reload-once`
    - In that special case when you want to simply reload the build you currenty have in `public` folder.
- Do not truncate objects in logger.
    - Now we use `util.inspect` to show objects in its full glory. Free color support too.
- Logger will now send objects with circular references without crashing.
- Report rendering performance when in dev mode.
- Enable debug logging when launching dev server with `--debug` flag.
- Add support for [Babel macros](https://github.com/kentcdodds/babel-plugin-macros).
- Implement a `multiline` macro for creating long multiline strings are unindented and compactified at compile-time.
- Rename interfaces to better match the component naming convention.
- Fix LabeledList.Divider.
- Sort routes by name.
- Remove `package-lock.json` files on `--clean` (we treat them as garbage anyway)
- Catch weird JSON parsing errors for better debuggability.
  - I suspect BYOND encodes `NaN` and `1.#IND` into JSON which are invalid values.
- Bump NodeJS version to v12.x on Travis. v8.x is reaching end of life soon.
- Add React ESLint plugin.
  - Set it up with sane defaults, keep the code clean and bug free.
  - Massive cleanup afterwards.

## Why It's Good For The Game

Improved developer experience.

## Changelog
:cl:
tweak: Lots of code improvements for tgui-next
/:cl:

### CODE REVIEW

**Comment by stylemistake** [comment]
Added some more things, found no bugs on a local server.

**Comment by stylemistake** [comment]
Rebased on top of Coin Press UI. Added eslint react plugin, since linter wasn't checking for react bugs and code style. Still ready to merge.

**Review by actioninja — CHANGES_REQUESTED** [review]
tiny complaint

File: tgui-next/packages/tgui-dev-server/link/client.js [hunk 1]
```diff
@@ -32,9 +32,28 @@ if (process.env.NODE_ENV !== 'production') {
...
+    if (typeof value === 'object' && value !== null) {
+      if (cache.indexOf(value) !== -1) {
+        return '[circular ref]';
+      }
+      cache.push(value);
+      return value;
+    }
+    return value;
+  });
+  cache = null;
```
[the remark this answers is not in the document]
Comment (stylemistake) [reply tgui-next/packages/tgui-dev-server/link/client.js]: This is to improve garbage collection, especially on IE8.

**Pull request merged**